app/(storyblok)/[[...slug]]/page.tsx

```diff
@@ -119,6 +128,64 @@ async function getStoryData(params: { slug: string[] }) {
   return { data: 404 };
 };
 
+/**
+ * Get a list of stories that are of component sbStoryMvp in reverse chronological order.
+ * This is used for the story filter pages.
+ */
+async function getStoryList(params: { slug: string[] }) {
```

Comment:
You probably don't have to do the fetching here. You could move the data fetching into the storyFilterPage component, or make it a sub-component of, and do the fetching there to keep this component tidy. That could possibly also solve the editor duplicative code issue.

Reply:
Have a look at https://nextjs.org/docs/app/building-your-application/rendering/composition-patterns#interleaving-server-and-client-components for examples and inspiration.

Reply:
@sherakama I actually did think about going that route but thought I'd do it in page since Storyblok is already initialized there 😂  If I go for the server component, do I just initialize using `StoryblokInit` again there 🤔 ?

Reply:
I don't think so, but I have run into issues with SB before. I'm pretty sure all you need is the API client 

```
const storyblokApi: StoryblokClient = getStoryblokApi();
```

If that has issues you can instantiate the API client yourself and get going.

Reply:
Oh I didn't realize! I'll try that first. For some reason I thought every time I use getStoryblokApi I need to pass it the access token again so that was part of why I originally did everything in page.tsx 😂 Thanks! I'll definitely give this a go and see if I run into any issues.

Reply:
It does looks like I need to pass it the access token again in order to fetch data in a server component. Not as straightforward as I thought 😂